src/mqtt_client.rs

```diff
@@ -385,6 +386,51 @@ impl<
         Ok(())
     }
 
+    /// Reply to a previously-received message.
+    ///
+    /// # Generics
+    /// * `N` specifies the maximum number of properties that the reply can contain.
+    ///
+    /// # Args
+    /// * [`ReplyOptions`] - Options used for constructing the reply.
+    /// * `default_response` - An optional default response topic to reply to.
+    /// * `data` - The data containing the response.
+    /// * `qos` - The QoS to transmit the response at.
+    /// * `retain` - The retain state of the response.
+    /// * `properties` - The properties to attach to the response.
```

Comment:
Note that it will copy correlation data and that `N` needs to be (at least but preferrably exactly) `properties.len() + 1`. Or see below.